lib/fog/azurerm/models/network/network_security_group.rb

```diff
@@ -48,6 +48,32 @@ def destroy
           service.delete_network_security_group(resource_group, name)
         end
 
+        def update(hash = {})
+          raise('Provided hash can not be empty.') if hash.empty? || hash.nil?
+
+          if !hash[:name].nil? || !hash[:location].nil? || !hash[:resource_group].nil? || !hash[:id].nil?
+            raise('You can not modify id:, name:, location: or resource_group:')
+          end
+          unless hash[:security_rules].nil?
+            validate_security_rules(hash[:security_rules])
+            merge_attributes(hash)
+            nsg = service.create_network_security_group(resource_group, name, location, security_rules)
```

Comment:
rename create_network_security_group to "create_or_update_network_security_group".

Can we implement this as update_security_rule?